lib/actions/ssh.ts

```diff
@@ -80,30 +79,29 @@ async function getContainerId(
 		// container
 		const { child_process } = await import('mz');
 		const escapeRegex = await import('lodash/escapeRegExp');
-		const { getSubShellCommand } = await import('../utils/helpers');
+		const { which } = await import('../utils/helpers');
 		const { deviceContainerEngineBinary } = await import('../utils/device/ssh');
 
-		const command = generateVpnSshCommand({
+		const sshBinary = await which('ssh');
+		const sshArgs = generateVpnSshCommand({
 			uuid,
 			verbose: false,
 			port: sshOpts.port,
-			command: `host ${uuid} '"${deviceContainerEngineBinary}" ps --format "{{.ID}} {{.Names}}"'`,
+			command: `host ${uuid} "${deviceContainerEngineBinary}" ps --format "{{.ID}} {{.Names}}"`,
 			proxyCommand: sshOpts.proxyCommand,
 			proxyUrl: sshOpts.proxyUrl,
 			username: sshOpts.username,
 		});
 
-		const subShellCommand = getSubShellCommand(command);
-		const subprocess = child_process.spawn(
-			subShellCommand.program,
-			subShellCommand.args,
-			{
-				stdio: [null, 'pipe', null],
-			},
-		);
+		if (process.env.DEBUG) {
+			console.error(`[debug] [${sshBinary}, ${sshArgs.join(', ')}]`);
+		}
+		const subprocess = child_process.spawn(sshBinary, sshArgs, {
+			stdio: [null, 'pipe', null],
```

Comment:
The documentation says that `null` defaults to `pipe`, so presumably `[null, 'pipe', null]` means the same as `pipe`. But I also seem to remember we had found corner cases on Windows (?) with certain versions of Node (?), so I thought I'd better leave it alone.